### ISSUE TITLE
Adding a new fluid and compiled it. Not working when function is used.

### ISSUE THREAD

**Issue by csm9420** [CoolProp/CoolProp#2067]
Hello, everyone. 

I am having a problem with the CoolProp and tried to find a way but failed. 
I hope that I can seek some guidelines in the forum.

So I added the new fluid, Hydrogen Peroxide, into the code. 
I accessed the pureFluids.py and added a new fluid with all the information inside. 
I checked that the json file was properly generated.

I tried to go to the next procedure, which seems to run the setup.py file in the wrappers folder. 
The code was conducted properly, so I hoped that I could now use my new fluid in the python.

However, when I type 

import CoolProp.CoolProp as CP
CP.PropsSI('D','P',1e5,'T',300,'H2O2')

the error comes  out saying that  

Traceback (most recent call last):
  File "<input>", line 1, in <module>
  File "CoolProp\CoolProp.pyx", line 384, in CoolProp.CoolProp.PropsSI
  File "CoolProp\CoolProp.pyx", line 464, in CoolProp.CoolProp.PropsSI
  File "CoolProp\CoolProp.pyx", line 351, in CoolProp.CoolProp.__Props_err2
ValueError: Initialize failed for backend: "?", fluid: "H2O2" fractions "[ 1.0000000000 ]"; error: key [H2O2] was not found in string_to_index_map in JSONFluidLibrary : PropsSI("D","P",100000,"T",300,"H2O2")

which seems that my newly added fluid is not in the library.
I am not sure where I lost myself. Should  I compile in different ways rather than executing setup.py?

Thank you for your help in advance.
Hope you all have a nice day.

**Comment by csm9420** [comment]
For additional information, I created venv in the folder where my main code  is located.
and I have my CoolProp-master folder in the desktop and configured the interpreter to the venv in the main code folder. 
After I executed setup.py, "\venv\Lib\site-packages\CoolProp-6.4.2.dev0-py3.7-win-amd64.egg" folder was created, with coolprop folder inside it.

**Comment by ibell** [comment]
Can you please show the generated Json? If an incompressible, make sure you
prefix the name with INCOMP::

**Comment by csm9420** [comment]
> Can you please show the generated Json? If an incompressible, make sure you prefix the name with INCOMP::

{
  "T_freeze": {
    "NRMS": null,
    "coeffs": "null",
    "type": "notdefined"
  },
  "Tbase": 360.0,
  "Tmax": 420.0,
  "Tmin": 300.0,
  "TminPsat": 300.0,
  "conductivity": {
    "NRMS": 0.0034421096970797117,
    "coeffs": [
      [
        0.6287566433566436
      ],
      [
        0.0003481481018981297
      ],
      [
        -4.2217782217781425e-06
      ],
      [
        8.828671328676898e-09
      ]
    ],
    "type": "polynomial"
  },
  "density": {
    "NRMS": 0.00021132964626024563,
    "coeffs": [
      [
        1370.2510489510494
      ],
      [
        -1.3946503496504392
      ],
      [
        -0.006948551448551221
      ],
      [
        -5.244755244770971e-05
      ]
    ],
    "type": "polynomial"
  },
  "description": "sc",
  "mass2input": {
    "NRMS": null,
    "coeffs": "null",
    "type": "notdefined"
  },
  "mole2input": {
    "NRMS": null,
    "coeffs": "null",
    "type": "notdefined"
  },
  "name": "H2O2",
  "reference": "sc",
  "saturation_pressure": {
    "NRMS": 0.15090390041105772,
    "coeffs": [
      -298.49932286490116,
      -0.6852848234454553,
      6.031235454437635
    ],
    "type": "logexponential"
  },
  "specific_heat": {
    "NRMS": 0.03530942610632857,
    "coeffs": [
      [
        3096.324475524477
      ],
      [
        0.49631368631373496
      ],
      [
        0.00872977022977034
      ],
      [
        3.0419580419549508e-05
      ]
    ],
    "type": "polynomial"
  },
  "viscosity": {
    "NRMS": 0.04754715731258706,
    "coeffs": [
      [
        -7.607884503729053
      ],
      [
        -0.013156535778407837
      ],
      [
        0.0001287578227829999
      ],
      [
        1.8414258679665813e-06
      ]
    ],
    "type": "exppolynomial"
  },
  "volume2input": {
    "NRMS": null,
    "coeffs": "null",
    "type": "notdefined"
  },
  "xbase": 0.0,
  "xid": "pure",
  "xmax": 1.0,
  "xmin": 0.0
}

**Comment by csm9420** [comment]
Should I change the name to INCOMP::H2O2 ??

**Comment by ibell** [comment]
Yes,  that's right. Another thing you might run into is that your new
coolprop can be shadowed by existing version,  so make sure you uninstall
coolprop until you can't anymore before installing.

**Comment by csm9420** [comment]
I am new to python so this may sound a bit stupid...  but if I make a new project with empty venv folder, doesn't that make the same effect of erasing all existing coolprop for my code?
and one more thing, if I follow the manual installation in the link (https://coolprop.dreamhosters.com/binaries/sphinx/coolprop/wrappers/Python/index.html#manual-installation)
after executing code "git clone https://github.com/CoolProp/CoolProp --recursive", I have CoolProp folder within the CoopProp master folder. Then, I end up with 2 dev folders. Which One Should I change?

**Comment by csm9420** [comment]
> Yes, that's right. Another thing you might run into is that your new coolprop can be shadowed by existing version, so make sure you uninstall coolprop until you can't anymore before installing.

Just for reminder, if I use the name to INCOMP::H2O2, this error pops out:

\CoolProp-master\dev\incompressible_liquids\CPIncomp\WriterObjects.py", line 413, in toJSON
    with open(path, 'w') as fp:
error: [Errno 22] Invalid argument: 'json\\INCOMP::H2O2.json'

**Comment by csm9420** [comment]
Thank you ibell for helping me out! I finally found out what mistake I made!
I will make a short description so that I can help someone like me in the future. 

So as I mentioned, I was able to create the json file after running the all_incompressibles file.
I compiled the code by running setup.py file at \wrappers\Python folder.

So I thought initially that I need to name the fluid as INCOMP::NAME in the pureFluids.py code located at dev\incompressible_liquids\CPIncomp folder, where NAME is the name of my new fluid (let say H2O2 in this case). 
But when I tried to name the fluid as mentioned, it resulted in an error as shown below:

\CoolProp-master\dev\incompressible_liquids\CPIncomp\WriterObjects.py", line 413, in toJSON
with open(path, 'w') as fp:
error: [Errno 22] Invalid argument: 'json\INCOMP::H2O2.json'

Here, I made mistakes. I needed to name the fluid H2O2 (at pureFluids.py code). 
Then, I'd need to recompile the code by running setup.py code located at CoolProp\wrappers\Python folder. 
I completed the compiling, I need to use INCOMP:: prefix to call the fluid for the function. Like so:

CP.PropsSI('D','P',1e5,'T',300,'INCOMP::H2O2')

So long story short, you can name the name you want at the pureFluids.py file, but you need to add a certain prefix to specify the name of the species when using the function.
Please comment below if you find anything wrong with my guidelines 👍

**Comment by jowr** [comment]
Thank you for the description - I am going to add this to the docs. Can you share your data with us for the next release?

**Comment by twea64** [comment]
Is this fluid available for others to use? Or is it possible to pull csm9420s version?

**Comment by csm9420** [comment]
> Is this fluid available for others to use? Or is it possible to pull csm9420s version?

Hi, twea64.

I'm not sure if this file is officially released for others to use. But, I manually created the file to include liquid Hydrogen Peroxide into the CoolProp and it worked all fine. I compared and validated the simulated results with the actual experiment as well.

**Comment by twea64** [comment]
Thanks for the reply csm9420! What concentration H2O2 was your model and experimental comparison? I would be keen to see this added to the official release if that's possible, otherwise, I will attempt to add it also.
